base/Plain_netCDF_Time.F90

```diff
@@ -441,12 +440,17 @@ subroutine parse_timeunit_i4(tunit, n, t0, dt, rc)
     read(s1, '(i4,a1,i2,a1,i2)') y, c1, m, c1, d
     read(s2, '(i2,a1,i2,a1,i2)') hour, c1, min, c1, sec
 
-    _ASSERT(trim(s_unit) == 'seconds', "s_unit /= 'seconds' is not handled")
-    isec=n
+    if (trim(s_unit) == 'seconds') then
+       isec=n
+    elseif (trim(s_unit) == 'minutes') then
+       isec=n * 60
+    elseif (trim(s_unit) == 'hours') then
+       isec=n * 3600
+    else
+       _FAIL ('time_unit not implemented')
+    end if
```

Comment:
This logic is duplicated below for the I8 interface.   Couldn't the I4 interface just call the I8 interface, e.g.:
```f90
integer(kind=ESMF_KIND_I8) :: n8
n8 = n
call parse_timeunit(tunit, n8, t0, dt, _RC)
```

Reply:
Thanks for pointing out this approach, which is much better!